src/stage1/analyze.cpp

```diff
@@ -5917,8 +5917,6 @@ static bool return_type_is_cacheable(ZigType *return_type) {
 }
 
 bool fn_eval_cacheable(Scope *scope, ZigType *return_type) {
-    if (!return_type_is_cacheable(return_type))
-        return false;
```

Comment:
This prevented the compiler from caching functions that returns slices and the likes. It seems to have been put in place to fix a bug 3 years ago, with at test case in `test/behavior/bugs/828.zig`. I've run the compiler on this case at it still succeeds, so this must then have been fixed some other way in the compiler since then